pubtools/pulplib/_impl/client/repo_lock.py

```diff
@@ -0,0 +1,161 @@
+import json
+import random
+import string
+from time import sleep
+
+from datetime import datetime, timedelta
+from more_executors.futures import f_map
+
+
+LOCK_CLAIM_STR = "lock-claim-"
+
+LOCK_VALID_FROM_OFFSET = 10
+LOCK_EXPIRATION_OFFSET = 3600
+
+# How long to sleep before checking if we can activate the lock again
+# This should be a sleep per locks ahead of us. If there's 3 locks ahead of
```

Comment:
> This should be a sleep per locks ahead of us

I feel like this part of the logic is unnecessary and might be better off being dropped.
I guess the benefit is only to reduce the number of requests to the server, but it seems negligible.

On the downside seems likely to introduce unnecessary delays.  I'd expect in practice that clients often need to hold the lock for a fair bit less than 15 seconds, so this could easily introduce a half minute delay where it's just sleeping even though everyone else already released their locks.